### PR TITLE
Added Favourites section

### LINKED ISSUE #563: Add Favourites section on `Home Page` at the top
- Add a new section on home page - `Favourites`
- It should be empty initially, Just show an icon (or animation) to allow users to add new items there
- When user clicks on the icon, open up a dialog with multiple selectable checkboxes. They should have all the item names that are there on the home page
- The items should then be shown in favourites
- These should be stored in shared prefs

### PR DESCRIPTION
**Pull request by jddeep**
# Description.
screenshot_20190307-210426

screenshot_20190307-210430

Fixes #563 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
Just one query:
- how are you deciding the layout of cards in the favourites section? I can see 1 card in first row & 2 in second row

File: .github/issue_template.md [hunk 1]
```diff
@@ -1,13 +0,0 @@
-Describe your issue here.
```
Comment (Swati4star) [inline .github/issue_template.md]: why delete this file?

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        ActionBar actionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
+        actionBar.hide();
+        mFab.setOnClickListener(v ->
+                startActivity(new Intent(this.getContext(), FavouritesActivity.class))
+        );
+        return rootview;
+    }
+
+    private void checkFavs(SharedPreferences sharedPreferences) {
+        if (sharedPreferences.getBoolean(getResources().getString(R.string.img_to_pdf),
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: too much code repitition here. Why dont you create a function that takes the required parameters and do this task. Will make the code easy to understand also

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                false)) {
+            pref_pdf_to_img.setVisibility(View.VISIBLE);
+        } else {
+            pref_pdf_to_img.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: remove this extra line

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        sharedPreferences.registerOnSharedPreferenceChangeListener(this);
+        ActionBar actionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
+        actionBar.hide();
+        mFab.setOnClickListener(v ->
+                startActivity(new Intent(this.getContext(), FavouritesActivity.class))
+        );
+        return rootview;
+    }
+
+    private void checkFavs(SharedPreferences sharedPreferences) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: add javadoc style comments to this function

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java [hunk 5]
```diff
@@ -201,7 +201,6 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     }
 
 
-
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: remove this file from commit

File: app/src/main/res/values-es/strings.xml [hunk 6]
```diff
@@ -377,7 +377,7 @@
     <string name="invert_pdf_info">All the colors in the pdf will be inverted</string>
     <string name="invert_pdf_title">Invert PDF</string>
     <string name="snackbar_invert_unsuccessful">The PDF could not be inverted</string>
-    <string name="snackbar_invert_successfull">Inverted</string>
+    <string name="snackbar_invert_successfull">Colors inverted successfully.</string>
```
Comment (Swati4star) [inline app/src/main/res/values-es/strings.xml]: you have reverted someone else's change here. look into this

File: app/src/main/res/values-fr/strings.xml [hunk 7]
```diff
@@ -377,7 +378,7 @@
     <string name="invert_pdf_info">All the colors in the pdf will be inverted</string>
     <string name="invert_pdf_title">Invert PDF</string>
     <string name="snackbar_invert_unsuccessful">The PDF could not be inverted</string>
-    <string name="snackbar_invert_successfull">Inverted</string>
+    <string name="snackbar_invert_successfull">Colors inverted successfully.</string>
```
Comment (Swati4star) [inline app/src/main/res/values-fr/strings.xml]: look into this

**Comment by vyankatesh24** [comment]
@jddeep why are you deleting issue_template.md file?

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 8]
```diff
@@ -2,11 +2,15 @@
...
 import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentManager;
 import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: Remove these imports. Unnecessary change.

File: app/src/main/res/layout/favourites_fragment.xml [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+                <swati4star.createpdf.customviews.MyCardView
+                    android:id="@+id/pdf_to_images_fav"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1.0"
+                    android:visibility="gone"
+                    app:option_icon="@drawable/ic_image_black_24dp"
+                    app:option_text="@string/pdf_to_images" />
+
+
```
Comment (srv-twry) [inline app/src/main/res/layout/favourites_fragment.xml]: Too many blank lines. Remove them.

File: app/src/main/res/layout/favourites_fragment.xml [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+            </LinearLayout>
+
+        </LinearLayout>
+
+    </ScrollView>
```
Comment (srv-twry) [inline app/src/main/res/layout/favourites_fragment.xml]: Remove these blank lines.

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 11]
```diff
@@ -82,6 +82,7 @@
             </menu>
         </item>
         <item android:title="@string/more_options">
+
```
Comment (srv-twry) [inline app/src/main/res/menu/activity_main_drawer.xml]: why a blank line?

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 4 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: @Swati4star you mean double slash comments above the function?

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 2 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: @Swati4star I thought about it but then I need to create Hash maps and run throughout it in a loop as I need to check for every checked/unchecked favourite

**Comment by jddeep** [comment]
Okay I will look into all changes. @Swati4star please tell what do you think about the replies which I gave to some above comments. Thanks

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 4 above)
Comment (srv-twry) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: @jddeep Nope. Move your cursor to line 86 in Android studio and press `/**` and then Enter. You'll automatically get the template for the Javadoc style comment.

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 2 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: @Swati4star @srv-twry Any thoughts on my above comment?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: You can move this part in a function, which takes stringid and view as parameteres 
```
if (sharedPreferences.getBoolean(getResources().getString(stringid),
                false)) {
            view.setVisibility(View.VISIBLE);
        } else {
            view.setVisibility(View.GONE);
        }
```
Now, if you wish you can call function for every view or create a simple array of views & string ids, and just traverse those

Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: @Swati4star Can you please tell me your thoughts about the above idea?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: I have already replied and given you an alternate to hashmap

Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: Okay @Swati4star . So then that function needs to be called multiple times right?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: yes. 

> call function for every view or create a simple array of views & string ids, and just traverse those

**Comment by jddeep** [comment]
@Swati4star Please review now.

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+import butterknife.ButterKnife;
+import swati4star.createpdf.R;
+import swati4star.createpdf.activity.FavouritesActivity;
+import swati4star.createpdf.customviews.MyCardView;
+
+public class FavouritesFragment extends Fragment
+        implements SharedPreferences.OnSharedPreferenceChangeListener {
+
+    private Activity mActivity;
+    private SharedPreferences mSharedPreferences;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: we already have this instance of SharedPreferences. Use this. Don't create new variable for this

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+                getResources().getString(R.string.pdf_to_images));
+    }
+
+    /**
+     * This method toggles the visibility of the passed view.
+     * @param sharedPreferences
+     * @param view
+     * @param id
+     */
+    private void viewVisibility(SharedPreferences sharedPreferences, View view, String id) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: no need to receive SharedPreferences parameter

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -140,7 +141,7 @@
     <string name="snackbar_no_duplicate_pdf">Duplicate pages not found.No new PDF created.</string>
     <string name="snackbar_duplicate_removed">A new PDF is created after removal of duplicate pages.</string>
     <string name="snackbar_invert_unsuccessful">The PDF could not be inverted</string>
-    <string name="snackbar_invert_successfull">Inverted</string>
+    <string name="snackbar_invert_successfull">Colors inverted successfully.</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: why this change?

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -85,6 +85,7 @@
     <string name="successToast">Images successfully added.</string>
 
     <!-- Navigation drawer strings -->
+    <string name="img_to_pdf">Images to PDF</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: why is this change required?

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 12 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: Yeah will remove it.

File: app/src/main/res/values/strings.xml (hunk 15 above)
Comment (jddeep) [reply app/src/main/res/values/strings.xml]: Because this is used as a key

File: app/src/main/res/values/strings.xml (hunk 14 above)
Comment (jddeep) [reply app/src/main/res/values/strings.xml]: Will remove it.

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 13 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: Actually we need it to get the boolean value for a given id

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+    @BindView(R.id.pdf_to_images_fav)
+    MyCardView pref_pdf_to_img;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootview = inflater.inflate(R.layout.favourites_fragment, container, false);
+        ButterKnife.bind(this, rootview);
+        SharedPreferences sharedPreferences = PreferenceManager
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: instead of creating `sharedPreferences`, I asked you to use previous `mSharedPreferences` variable. You have deleted it.

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 13 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: You will directly be able to use `mSharedPreferences` variable. There wont be any need to get this as a parameter

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 16 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: I understood the other way around.

File: app/src/main/res/values/strings.xml (hunk 15 above)
Comment (Swati4star) [reply app/src/main/res/values/strings.xml]: We dont need translations of keys, so we add keys in a constants file.
https://github.com/Swati4star/Images-to-PDF/blob/master/app/src/main/java/swati4star/createpdf/util/Constants.java

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 13 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: Yeah I got it now. I had misunderstood you.

File: app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java (hunk 12 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java]: You werent supposed to remove `mSharedPreferences`

File: app/src/main/res/values/strings.xml (hunk 15 above)
Comment (jddeep) [reply app/src/main/res/values/strings.xml]: Okay will change it then.

**Comment by jddeep** [comment]
@Swati4star Do tell if there are any other changes. I will make a final commit tonight. Thanks

**Comment by Swati4star** [comment]
I already mentioned all the required changes.

**Comment by jddeep** [comment]
Yeah other than those.

**Comment by jddeep** [comment]
@Swati4star All changes have been seen into. Thanks!

**Comment by Swati4star** [comment]
Awesome @jddeep 👍 
Took a lot of iterations! Good job on accommodating all the changes.
